File: 9-palindrome-number/9-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x < 0 || (x%10 == 0 && x != 0)) return false;
        
        int a = 0;
        
        while(x > a){
            a = 10*a + (x%10);
            x = x/10;
        }
        
        return (x == a || x == a/10) ;
    }
}

File: 9-palindrome-number/9-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x < 0) return false;
        long a = x;
        long b = 0;
        
        while(x > 0){
            int r = x%10;
            b = 10*b + r;
            x = x/10;
        }
        
        if(a == b) return true;
        return false;
    }
};